src/utils/plotSpecs/generateMitochondrialContentSpec.js

```diff
@@ -139,4 +139,56 @@ const generateSpec = (config, plotData) => {
   };
 };
 
-export default generateSpec;
+const filterCells = (cellSets, selectedCellSet) => {
+  let newCellSets = cellSets.hierarchy.find(
+    (rootNode) => rootNode.key === selectedCellSet,
+  )?.children || [];
+
+  // Build up the data source based on the properties. Note that the child nodes
+  // in the hierarchy are /objects/ with a `key` property, hence the destructuring
+  // in the function.
+  newCellSets = newCellSets.flatMap(({ key }) => {
+    const cells = Array.from(cellSets.properties[key].cellIds);
+
+    return cells.map((cellId) => ({
+      cellId,
+      cluster: cellSets.properties[key].name,
```

Comment:
Are we using `cluster` and `color` ? `generateData` seems to be ignoring them...

Reply:
No, they aren't used. I'll remove them